### DISCUSSION TITLE
"Closest interpolation mode"?

### DISCUSSION THREAD

**Discussion by ghost** [bnpr/Malt#197] [Q&A] [answered]
Is there a way to use a texture in Malt and have it set to the "closest interpolation mode"? I use pixel art textures and, they get blurred to hell.
![image](https://user-images.githubusercontent.com/96006654/145711636-f486e946-0c5b-47e9-b086-f8e82deebb2f.png)

**Comment by pragma37** [comment, marked as answer]
I added a sampler2D_sample_nearest function to the node utils.

So you can update the latest release, or simply add this function to your library:
```glsl
vec4 sampler2D_sample_nearest(sampler2D t, vec2 uv){ return texelFetch(t, ivec2(textureSize(t, 0) * uv), 0); }
```

**Reply by ghost** [reply]
Wow, that was fast! Thanks!

**Reply by ghost** [reply]
You should probably include a beer example in the "From Nodes To Code" thingy. Here's what I came up with thanks to you!
```
#include "Pipelines/NPR_Pipeline.glsl"

uniform int uv_channel = 0;
uniform sampler2D color_texture;

void COMMON_PIXEL_SHADER(Surface S, inout PixelOutput PO)
{
  vec2 texture_coordinates = S.uv[uv_channel];
  vec4 sampled_color = texelFetch(color_texture, ivec2(textureSize(color_texture, 0) * texture_coordinates), 0);
  
  PO.color = sampled_color;
}
```
